src/plugins/intel_npu/src/plugin/npuw/weights_bank.cpp

```diff
@@ -0,0 +1,46 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "weights_bank.hpp"
+
+using ov::npuw::weights::Bank;
+using ov::npuw::weights::BankManager;
+
+Bank::Bank(const std::string& device_name) : m_device_name(device_name) {
+    if (m_device_name != "CPU") {
+        OPENVINO_THROW("NPUW doesn't support ", m_device_name, " device for weights sharing!");
+    }
+}
+
+ov::Tensor Bank::get(const ov::Tensor& tensor) {
+    if (!tensor) {
+        OPENVINO_THROW("Uninitialized tensor in weights bank allocation!");
+    }
+
+    void* host_data_ptr = tensor.data();
+
+    std::lock_guard<std::mutex> guard(m_mutex);
+
+    if (m_bank.find(host_data_ptr) == m_bank.end()) {
+        // need to allocate first
+        m_bank[host_data_ptr] = tensor;
+    }
+
+    return m_bank[host_data_ptr];
```

Comment:
you didn't save one more lookup here - just store the result of `find` and use that iterator.

Reply:
Returning tensor now